src/tests/Interop/COM/ComWrappers/API/Program.cs

```diff
@@ -188,9 +189,14 @@ public void ValidateComInterfaceCreationRoundTrip()
             var testObjUnwrapped = wrappers.GetOrCreateObjectForComInstance(comWrapper, CreateObjectFlags.Unwrap);
             Assert.Same(testObj, testObjUnwrapped);
 
+            // UniqueInstance and Unwrap should always be a new com object, never unwrapped
+            var testObjUniqueUnwrapped = wrappers.GetOrCreateObjectForComInstance(comWrapper, CreateObjectFlags.Unwrap | CreateObjectFlags.UniqueInstance);
+            Assert.NotSame(testObj, testObjUniqueUnwrapped);
```

Comment:
Why not bring back the assert for `ComObject`?

Reply:
The ComWrappers implementation returns an `ITestObjectWrapper`. I suppose it probably should check to make sure it is that type rather than the original object.

Reply:
Oh right. This is doing everything manually. Thanks.